uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -250,25 +274,45 @@ define([
         vm.error = undefined;
       };
 
-      (function () {
-        // Create an empty leave request
-        vm.leaveRequest = LeaveRequestInstance.init({
-          contact_id: contactId //resolved from directive
-        }, false);
+      /**
+       * initializes the controller on loading the dialog
+       */
+      (function initController() {
+        if (directiveOptions.leaveRequest) {
+          //manager is responding to leave request
+          vm.uiOptions.isManager = directiveOptions.leaveRequest.contact_id != directiveOptions.contactId;
+          //owner is editing popup
+          vm.uiOptions.isOwner = directiveOptions.leaveRequest.contact_id == directiveOptions.contactId;
+        }
+
+        if (canEdit()) {
```

Comment:
Why can't you clone the leave request anyway, if it has been passed? Why do you need to check first if the user can edit? It seems unnecessary logic to me. Even if we were in "view" mode, cloning the leave request would not make any difference, and you would streamline the code logic